src/tickit/core/management/event_router.py

```diff
@@ -194,6 +198,18 @@ def input_components(self) -> Set[ComponentID]:
             for dev, _ in port
         )
 
+    @cached_property
+    def isolated_components(self) -> Set[ComponentID]:
+        """A cached set of components without inputs or outputs.
+
+        Returns:
+            Set[ComponentID]: A set of components which are isolated
+        """
+        return set.difference(
+            {component for component, _ in self.wiring.items()},
```

Comment:
Think you could also have 
```suggestion
            set(self.wiring.keys()),
```
Less verbose but perhaps less clear?